benchmark/src/context-binding/context-binding.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright IBM Corp. 2018,2020. All Rights Reserved.
+// Node module: @loopback/benchmark
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Context, inject, Provider, ValueFactory} from '@loopback/context';
+import Benchmark from 'benchmark';
+
+/**
+ * First option - use a factory function
+ */
+const factory: ValueFactory = ({context}) => {
+  const user = context.getSync('user');
+  return `Hello, ${user}`;
+};
+
+/**
+ * First option - use a factory function
+ */
+const asyncFactory: ValueFactory = async ({context}) => {
+  const user = await context.get('user');
+  return `Hello, ${user}`;
+};
+
+/**
+ * Third option - use a value factory provider class with static method
+ * parameter injection
+ */
+class StaticGreetingProvider {
+  static value(@inject('user') user: string) {
+    return `Hello, ${user}`;
+  }
+}
+
+class AsyncStaticGreetingProvider {
+  static value(@inject('user') user: string) {
+    return Promise.resolve(`Hello, ${user}`);
+  }
+}
+
+/**
+ * Fourth option - use a regular provider class
+ */
+class GreetingProvider implements Provider<string> {
+  @inject('user')
+  private user: string;
+
+  value() {
+    return `Hello, ${this.user}`;
+  }
+}
+
+class AsyncGreetingProvider implements Provider<string> {
+  @inject('user')
+  private user: string;
+
+  value() {
+    return Promise.resolve(`Hello, ${this.user}`);
+  }
+}
+
+setupContextBindings();
+
+function setupContextBindings() {
+  const ctx = new Context();
+  ctx.bind('user').to('John');
+  ctx.bind('greeting1').toDynamicValue(factory);
+  ctx.bind('greeting2').toDynamicValue(asyncFactory);
+  ctx.bind('greeting3').toDynamicValue(StaticGreetingProvider);
+  ctx.bind('greeting4').toDynamicValue(AsyncStaticGreetingProvider);
+  ctx.bind('greeting5').toProvider(GreetingProvider);
+  ctx.bind('greeting6').toProvider(AsyncGreetingProvider);
```

Comment:
Please use more descriptive binding names than `greeting1` to `greeting6`. For example:

```suggestion
  ctx.bind('greeting.AsyncGreetingProvider').toProvider(AsyncGreetingProvider);
```